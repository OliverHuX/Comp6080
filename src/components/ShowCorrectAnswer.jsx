import React from 'react';
import { Typography } from '@material-ui/core';
import PropTypes from 'prop-types';

export default function ShowCorrectAnswer ({ answer, answerIds }) {
  return <>
    <Typography component="h1" variant="h5" align="center" color="primary" gutterBottom>
      Answer:
      <br></br>
    </Typography>
    {answer.length !== 0 && answerIds.map(item => {
      return (
        <Typography component="h1" variant="h5" align="center" color="primary" gutterBottom key={item}>
          {answer && Object.keys(answer[item])}
        </Typography>
      )
    })}
  </>
}
ShowCorrectAnswer.propTypes = {
  answer: PropTypes.array,
  answerIds: PropTypes.array,
};
