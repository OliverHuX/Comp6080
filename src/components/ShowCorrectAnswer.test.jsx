import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ShowCorrectAnswer from './ShowCorrectAnswer';

describe('Title Component', () => {
  const data = [
    { answer1: true },
    { answer2: false },
    { answer3: false },
    { answer4: true },
    { answer5: false },
    { answer6: true },
  ]
  it('has default title', () => {
    const answer = shallow(<ShowCorrectAnswer answer={[]} ></ShowCorrectAnswer>);
    const title = answer.props().children[0].props;
    expect(title.children[0]).toBe('Answer:');
    expect(title.component).toBe('h1');
    expect(title.variant).toBe('h5');
    expect(title.align).toBe('center');
    expect(title.color).toBe('primary');
  });
  it('renders with one data', () => {
    const answer = shallow(<ShowCorrectAnswer answer={data.slice(0, 2)} answerIds={[0]} ></ShowCorrectAnswer>);
    expect(answer.props().children[1].length).toBe(1);
    const content1 = answer.props().children[1][0].props.children[0];
    expect(content1).toBe('answer1');
  });
  it('renders with multiple data', () => {
    const answer = shallow(<ShowCorrectAnswer answer={data} answerIds={[0, 3, 5]} ></ShowCorrectAnswer>);
    expect(answer.props().children[1].length).toBe(3);
    const content1 = answer.props().children[1][0].props.children[0];
    expect(content1).toBe('answer1');
    const content2 = answer.props().children[1][1].props.children[0];
    expect(content2).toBe('answer4');
    const content3 = answer.props().children[1][2].props.children[0];
    expect(content3).toBe('answer6');
  });
  it('renders with minimal props (snapshot)', () => {
    const answer = renderer.create(<ShowCorrectAnswer answer={[]} ></ShowCorrectAnswer>).toJSON();
    expect(answer).toMatchSnapshot();
  });
  it('renders with one data (snapshot)', () => {
    const answer = renderer.create(<ShowCorrectAnswer answer={data.slice(0, 2)} answerIds={[0]} ></ShowCorrectAnswer>).toJSON();
    expect(answer).toMatchSnapshot();
  });
  it('renders with multiple data (snapshot)', () => {
    const answer = renderer.create(<ShowCorrectAnswer answer={data} answerIds={[0, 3, 5]} ></ShowCorrectAnswer>).toJSON();
    expect(answer).toMatchSnapshot();
  });
})
