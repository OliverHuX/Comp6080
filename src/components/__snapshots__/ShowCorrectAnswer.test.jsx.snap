// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component renders with minimal props (snapshot) 1`] = `
<h1
  className="MuiTypography-root MuiTypography-h5 MuiTypography-colorPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
>
  Answer:
  <br />
</h1>
`;

exports[`Title Component renders with multiple data (snapshot) 1`] = `
Array [
  <h1
    className="MuiTypography-root MuiTypography-h5 MuiTypography-colorPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
  >
    Answer:
    <br />
  </h1>,
  <h1
    className="MuiTypography-root MuiTypography-h5 MuiTypography-colorPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
  >
    answer1
  </h1>,
  <h1
    className="MuiTypography-root MuiTypography-h5 MuiTypography-colorPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
  >
    answer4
  </h1>,
  <h1
    className="MuiTypography-root MuiTypography-h5 MuiTypography-colorPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
  >
    answer6
  </h1>,
]
`;

exports[`Title Component renders with one data (snapshot) 1`] = `
Array [
  <h1
    className="MuiTypography-root MuiTypography-h5 MuiTypography-colorPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
  >
    Answer:
    <br />
  </h1>,
  <h1
    className="MuiTypography-root MuiTypography-h5 MuiTypography-colorPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
  >
    answer1
  </h1>,
]
`;
